web/src/rendered_markdown.js

```diff
@@ -72,12 +72,26 @@ export function set_name_in_mention_element(element, name) {
     }
 }
 
+function is_client_safari() {
+    // Since GestureEvent is only supported on Safari, we can use it
+    // to detect if the browser is Safari including Safari on iOS.
+    // https://developer.mozilla.org/en-US/docs/Web/API/GestureEvent
+    return Boolean(window.GestureEvent);
+}
+
 export const update_elements = ($content) => {
     // Set the rtl class if the text has an rtl direction
     if (rtl.get_direction($content.text()) === "rtl") {
         $content.addClass("rtl");
     }
 
+    if (is_client_safari()) {
+        $content.find(".message_inline_video video").each(function () {
+            // Without this video thumbnail doesn't load on Safari.
+            this.load();
+        });
+    }
```

Comment:
What's the downside of this being called on other clients -- I guess it loads the video twice? 

Also is https://api.jquery.com/load/ what we're using here?

Reply:
Yes, to avoid loading video metadata twice.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/load is the function called here.